be/src/olap/compaction.cpp

```diff
@@ -529,8 +532,58 @@ Status Compaction::modify_rowsets(const Merger::Statistics* stats) {
             }
         }
 
+        // Here we will calculate all the rowsets delete bitmaps which are committed but not published to reduce the calculation pressure
+        // of publish phase.
+        // All rowsets which need to recalculate have been published so we don't need to acquire lock.
+        // Step1: collect this tablet's all committed rowsets' delete bitmaps
+        TxnManager::txn_tablet_map_t txn_tablet_map {};
```

Comment:
we'd better to move this part after L589, do these works with lock, which can avoid potential consistency issues